core/modules/field/field.install

```diff
@@ -122,19 +122,7 @@ function field_update_1002() {
  * Uninstall the options_element contrib module, now part of core.
  */
 function field_update_1003() {
-  $module = db_query("SELECT name FROM {system} WHERE name = 'options_element' AND type = 'module'")->fetchField();
-  if ($module) {
-    $path = backdrop_get_path('module', 'options_element');
-    backdrop_set_message(t('The contributed module <em>Options Element</em> has been located at %path. The module has been disabled, since its functionality is now provided by Backdrop core. It is recommended that you remove this module from your site.', array('%path' => BACKDROP_ROOT . '/' . $path)), 'warning');
-    // Remove the entry for the Options Element module from the system table. As
-    // this module does not provide any config or settings, there doesn't seem
-    // to be anything else left to clean up or migrate.
-    // While the contrib module remains in the filesystem, this entry will be
-    // recreated, but it will have status 0 (disabled), and schema -1
-    // (uninstalled), so it is safe to remove.
-    db_query("DELETE FROM {system} WHERE name = 'options_element' AND type = 'module'");
-  }
-  else {
+  if (!update_uninstall_merged_module('options_element')) {
     return t('<em>Options Element</em> not found. Nothing to be done.');
   }
```

Comment:
I find this to be weird that we return a message when nothing happens, but display nothing when we do something. I know there is the big message at the top of the update page, but seems like we should return a message in the update results area as well?

Reply:
Yeah, I thought of that, but in my head this works like this:

- if a module has been found and was disabled, then we need to warn the user and let them know that it is advised to remove the contrib module from the filesystem.
- if nothing was found, we just communicate this in a less prominent way, via the update results area.

The update results area has traditionally been used to add log-type errors (like SQL errors for example). We could deprecate it, and use `backdrop_set_message()` for everything(?).

In any way, we could return something like `'<em>Options Element</em> found and disabled.'`.

I think that this is another bit that could be included in `update_uninstall_merged_module()` ...I mean, we introduced this function in order to not be duplicating code, and now we are returning the "not found. Nothing to be done." message in each update hook. Unsure how to implement it properly though. What if I added another `&$update_results_message` parameter to it, which was passed by reference? ...we could then return that too. Would that work though? I need to experiment more 😅